e3sm_diags/driver/utils/regrid.py

```diff
@@ -367,6 +367,12 @@ def align_grids_to_lower_res(
     ds_a_new = ds_a.copy()
     ds_b_new = ds_b.copy()
 
+    # If grids are equal, then no regridding is required. This ensures no
+    # performance cost is incurred.
+    equal_grids = _are_grids_equal(ds_a_new, ds_b_new)
+    if equal_grids:
+        return ds_a_new, ds_b_new
+
```

Comment:
Some drivers call  `align_grids_to_lower_res()` directly and don't use `subset_and_align_datasets()` (which calls `align_grids_to_lower_res()`.

The equality check should live in `align_grids_to_lower_res()` in order for all relevant drivers to skip regridding if needed.

Reply:
I was condering include the checker in `align_grids_to_lower_res()` at the first place.  Is `align_grids_to_lower_res()` only does horizontal reg?  I think it is, if it is the case, adding the check here may be save more computation time.  Without your change, my zppy run results seem to produce complete and close results in model vs obs. I can run again with your change, to see if it is the same, and we get more saving.

Reply:
Yes, `align_grids_to_lower_res()` only does horizontal regridding. I expect the performance to be comparable to what you just in this branch before my commit since the same check occurs, just in another function.